zone/parcels.cpp

```diff
@@ -0,0 +1,755 @@
+				}
+
+				if (item->ID == PARCEL_MONEY_ITEM_ID) {
+					inst->SetPrice(p.second.quantity);
+					inst->SetCharges(1);
+				}
+
+				pms.player_name     = p.second.from_name;
+				pms.sent_time       = p.second.sent_date;
+				pms.note            = p.second.note;
+				pms.serialized_item = inst->Serialize(p.second.slot_id);
+				pms.slot_id         = p.second.slot_id;
+				ar(pms);
+
+				uint32 packet_size = ss.str().length();
+				auto   out         = new EQApplicationPacket(OP_ItemPacket, packet_size);
+				if (out->size != packet_size) {
+					LogError(
+						"Attempted to send a parcel packet of mismatched size {} with a buffer size of {}.",
+						out->Size(),
+						packet_size
+					);
+					return;
+				}
+				memcpy(out->pBuffer, ss.str().data(), out->size);
+				FastQueuePacket(&out);
+
+				ss.str("");
+				ss.clear();
+			}
+		}
+
+	}
+	if (m_parcels.size() >= PARCEL_LIMIT) {
+		LogError(
+			"Found {} parcels for Character {}.  List truncated to the PARCEL_MAX_ITEMS [{}] + 5.",
+			m_parcels.size(),
+			GetCleanName(),
+			PARCEL_MAX_ITEMS
+		);
+		SendParcelStatus();
+		return;
+	}
+}
+
+void Client::SendParcel(Parcel_Struct parcel_in)
+{
+	auto results = ParcelsRepository::GetWhere(
+		database,
+		fmt::format(
+			"`to_name` = '{}' AND `slot_id` = '{}' LIMIT 1",
+			Strings::Escape(parcel_in.send_to),
+			parcel_in.item_slot
+		)
+	);
+
+	if (results.empty()) {
+		return;
+	}
+
+	ParcelMessaging_Struct pms{};
+	pms.packet_type = ItemPacketParcel;
+
+	std::stringstream           ss;
+	cereal::BinaryOutputArchive ar(ss);
+
+	BaseParcelsRepository::Parcels parcel{};
+	parcel.from_name = results[0].from_name;
+	parcel.id        = results[0].id;
+	parcel.note      = results[0].note;
+	parcel.quantity  = results[0].quantity;
+	parcel.sent_date = results[0].sent_date;
+	parcel.item_id   = results[0].item_id;
+	parcel.slot_id   = results[0].slot_id;
+	parcel.to_name   = results[0].to_name;
+
+	auto item = database.GetItem(parcel.item_id);
+	if (item) {
+		auto inst = database.CreateItem(item, parcel.quantity);
+		if (inst) {
+			inst->SetCharges(parcel.quantity > 0 ? parcel.quantity : 1);
+			inst->SetMerchantCount(1);
+			inst->SetMerchantSlot(parcel.slot_id);
+			if (inst->IsStackable()) {
+				inst->SetCharges(parcel.quantity);
+			}
+
+			if (item->ID == PARCEL_MONEY_ITEM_ID) {
+				inst->SetPrice(parcel.quantity);
+				inst->SetCharges(1);
+			}
+
+			pms.player_name     = parcel.from_name;
+			pms.sent_time       = parcel.sent_date;
+			pms.note            = parcel.note;
+			pms.serialized_item = inst->Serialize(parcel.slot_id);
+			pms.slot_id         = parcel.slot_id;
+			ar(pms);
+
+			uint32 packet_size = ss.str().length();
+			auto   out         = new EQApplicationPacket(OP_ItemPacket, packet_size);
+			if (out->size != packet_size) {
+				LogError(
+					"Attempted to send a parcel packet of mismatched size {} with a buffer size of {}.",
+					out->Size(),
+					packet_size
+				);
+				return;
```

Comment:
Memory leak on `out` and `inst` if this returns

Reply:
As noted above, converted to unique_ptr.